mpc/src/honeybadger/mod.rs

```diff
@@ -1095,24 +1095,22 @@ where
 
                 try_join_all(triple_init_futures).await?;
 
-                // Collect all triple outputs
+                // Collect all triple outputs into a map keyed by session ID
+                // IMPORTANT: We must add outputs in deterministic session ID order, not arrival order.
+                // With network delays, different parties may receive outputs in different orders.
+                // If we add in arrival order, parties get mismatched triples causing interpolation failures.
                 let triple_channel = self.outputchannels.triple_channel.clone();
+                let mut collected_triples: HashMap<SessionId, Vec<ShamirBeaverTriple<F>>> = HashMap::new();
```

Comment:
HashMaps in Rust don't enforce an ordering. Shouldn't another data structure be used?